bazel/cc_toolchains/clang_cc_toolchain_config.bzl

```diff
@@ -222,17 +222,6 @@ def _impl(ctx):
                 ],
                 flag_groups = [flag_group(flags = ["-shared"])],
             ),
-            flag_set(
-                actions = [
-                    ACTION_NAMES.cpp_link_executable,
-                ],
-                flag_groups = [
-                    flag_group(
-                        flags = ["-pie"],
```

Comment:
My poking around in the Clang driver suggests:

-   On GNU platforms such as Linux, one uses `-fpie` or `fPIE` when compiling and `-pie` when linking; `-pie` is passed from Clang to the linker. `-fpie` or `-fPIE` has no effect when linking and `-pie` has no effect when compiling.
-   On MacOS, one uses `-fpie` or `-fPIE` both when compiling and when linking; Clang translates a driver `-fpie` or `-fPIE` flag into a linker `-pie` flag. `-pie` has no effect on compilation or linking.

Weird. But this suggests that what we should do on MacOS is to pass `-fpie` or `-fPIE` when linking instead of passing `-pie`. (Or I suppose we could use `-Wl,-pie` which should work for both.)

Reply:
@chandlerc objected to the `-Wl,-pie` approach at: https://github.com/carbon-language/carbon-lang/pull/1605#discussion_r928098938

How about the current approach?